parl/algorithms/torch/policy_gradient.py

```diff
@@ -0,0 +1,87 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+import torch.optim as optim
+import parl
+from torch.distributions import Categorical
+
+__all__ = ['PolicyGradient']
+
+
+class PolicyGradient(parl.Algorithm):
+    def __init__(self, model, lr):
+        """Policy gradient algorithm
+
+        Args:
+            model (parl.Model): model defining forward network of policy.
+            lr (float): learning rate.
+
+        """
+        assert isinstance(lr, float)
+
+        self.model = model
+        device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+        self.model.to(device)
+        self.optimizer = optim.Adam(self.model.parameters(), lr=lr)
+
+    def predict(self, obs):
+        """Predict actions of maximum probability
+
+        Args:
+            obs (torch.tensor): shape of (*, obs_dim)
```

Comment:
replace `*` with `batch_size`.

Reply:
But in the example, the shape of obs in predict/sample is (obs_dim). we don't need to expand dims to the shape of (1, obs_dim) in the torch.

Reply:
Alright, but `*` normally does not represent that it can be ignored here. Please update the comment as `(batch_size, obs_dim)` or `(obs_sim,)`

Reply:
ok